src/collective/solr/tests/test_solr.py

```diff
@@ -69,7 +69,19 @@ def test_connection_str(self):
             str(c),
             "SolrConnection{host=localhost:8983, solrBase=/solr/plone, "
             "persistent=True, postHeaders={'Content-Type': 'text/xml; "
-            "charset=utf-8'}, reconnects=0}",
+            "charset=utf-8'}, reconnects=0, login=None, password=None}",
+        )
+
+    def test_connection_str_authentication(self):
+        c = SolrConnection(
+            host="localhost:8983", persistent=True, login="login", password="password"
+        )
+        self.assertEqual(
+            str(c),
+            "SolrConnection{host=localhost:8983, solrBase=/solr/plone, "
+            "persistent=True, postHeaders={'Content-Type': 'text/xml; "
+            "charset=utf-8', 'Authorization': 'Basic bG9naW46cGFzc3dvcmQ='}, "
```

Comment:
I think decoding this header will expose the password... Do you think this is a big deal @mpeeters?

Reply:
```python
>>> import base64
>>> base64.decodebytes(b"bG9naW46cGFzc3dvcmQ=")
b'login:password'
```

Reply:
@ale-rt I don't think this is a problem.

FYI, the authorization header is encoded because this is required by SolR.

For the security part this is similar to smtp server password on Plone, the password can be retrieved by a developer using a view or anything that can access to registry or SolR connection.

Reply:
That is not the point, if you can access the code you can do funny stuff, we know that.
My point is that I do not see any obvious reason that add the password in the `str` representation of that connection.
This is not a threat by itself but that string might land in an unforeseen way in logs or on a page.

Anyway, not a blocker for me.

Reply:
I have adapted the code to obfuscate Authorization header that was displayed simply because all headers are displayed. Thanks for your feedback